packages/atomic/src/components/search/generated-answer/atomic-generated-answer-feedback/atomic-generated-answer-feedback-modal.tsx

```diff
@@ -0,0 +1,275 @@
+        </legend>
+        {options.map(({id, localeKey, correspondingAnswer}) => (
+          <div
+            class={`flex items-center ${correspondingAnswer}`}
+            key={id}
+            part="reason"
+          >
+            <input
+              part="reason-radio"
+              type="radio"
+              name="answer"
+              id={id}
+              checked={this.currentAnswer === correspondingAnswer}
+              onChange={(e) =>
+                (e.currentTarget as HTMLInputElement | null)?.checked &&
+                this.setCurrentAnswer(correspondingAnswer)
+              }
+              class="mr-2 w-4 h-4"
+              required
+            />
+            <label part="reason-label" htmlFor={id}>
+              {this.bindings.i18n.t(localeKey)}
+            </label>
+          </div>
+        ))}
+      </fieldset>
+    );
+  }
+
+  private renderDetails() {
+    if (this.currentAnswer !== 'other') {
+      return;
+    }
+
+    return (
+      <fieldset>
+        <textarea
+          part="details-input"
+          name="answer-details"
+          ref={(detailsInput) => (this.detailsInputRef = detailsInput)}
+          class="mt-2 p-2 w-full text-base leading-5 border border-neutral resize-none rounded"
+          rows={4}
+          placeholder={this.bindings.i18n.t(
+            'generated-answer-feedback-textarea-add-details'
+          )}
+          required
+        ></textarea>
+      </fieldset>
+    );
+  }
+
+  private renderBody() {
+    console.log('feedbackSubmitted', this.feedbackSubmitted);
+    return !this.feedbackSubmitted ? (
+      <form
+        part="form"
+        id={this.formId}
+        slot="body"
+        onSubmit={(e) => {
+          e.preventDefault();
+          this.sendFeedback();
+        }}
+        class="flex flex-col gap-8"
+      >
+        {this.renderOptions()}
+        {this.renderDetails()}
+      </form>
+    ) : (
+      <div slot="body" class="flex flex-col items-center">
+        <atomic-icon icon={Success} class="w-48" />
+        <p class="text-sm">
+          {this.bindings.i18n.t('generated-answer-feedback-success')}
+        </p>
+      </div>
+    );
+  }
+
+  private renderFooter() {
+    return (
+      <div slot="footer" part="modalFooter">
+        {!this.feedbackSubmitted ? (
+          <div part="buttons" class="flex justify-end gap-2 p-2">
+            <Button
+              part="cancel-button"
+              style="outline-neutral"
+              class="text-primary"
+              onClick={() => this.close()}
+            >
+              {this.bindings.i18n.t('cancel')}
+            </Button>
+            <Button
+              part="submit-button"
+              style="primary"
+              type="submit"
+              form={this.formId}
+              disabled={!this.currentAnswer ? true : false}
+              onClick={() => {
+                this.feedbackSubmitted = true;
```

Comment:
Is the feedback modal intended to be single-use? In other words, it seems that once a user submits feedback, opening the modal again would bring back the "success" page, and not the feedback form.

I'm not sure how likely it is for a user to submit more than one feedback for the same generated answer, but I feel it would make more sense to reset the modal once feedback is submitted.

Thoughts?

Reply:
Right now if the user submits, then re opens the modal, it starts over, letting them submit more feedback. I'm not exactly if it was supposed to be a single time only or not, @johnnyCoveo @nathanlb what do you think?

Reply:
In my opinion, users are expecting the modal to start clean every time it opens. If that is the actual behavior, then it's all good. I probably missed something when reviewing, will check again.

Reply:
I've done some testing and the modal state is currently not cleaned when the modal is closed. In other words, if you open the modal a second time, it remembers the option selected previously (even if feedback was submitted).

In my opinion, the modal should start clean every time it opens.